### ISSUE TITLE
Array and string offset validity doesn't recognize string returned by `mb_strtolower`

### ISSUE THREAD

**Issue by keulinho** [kalessil/phpinspectionsea#1427]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 3.0.4.2 |
| Language level | PHP 7.2                    |

Current behaviour
--

Passing an string as an array offset that is returned from `mb_strtolower` leads to an `Array and string offset validity` warning, message is `Resolved index type ([bool, array]) is incompatible with possible [string, int]. Probalbly just proper type hinting is needed`. 

Minimal code to reproduce:
```php
        $array = [];
        $test = mb_strtolower('test');
        $value = $array[$test];
```

Using `strtolower()` instead of `mb_strtolower()` leads to no warning.

Expected behaviour
--
No warning should be risen just like when using `mb_strtolower()`.

Environment details
---
PhpStorm 2019.2.3 on Ubuntu 16.04

**Comment by kalessil** [comment]
Thank you for reporting @keulinho, I was able to reproduce only with symfony polifills. 
The polifills seems to miss PhpDoc, so IDE solves polifill type and think that the function could return array.

On our end we can not fix this, perhaps a PR to symfony polifills could help (I'm not sure though if maintainers see the same way as me).
